Sources/AWSLambdaDeploymentDescriptor/DeploymentDefinition.swift

```diff
@@ -0,0 +1,526 @@
+            self.resources[res.name] = res
+        }
+    }
+    
+    enum CodingKeys: String, CodingKey {
+        case awsTemplateFormatVersion = "AWSTemplateFormatVersion"
+        case transform = "Transform"
+        case description = "Description"
+        case resources = "Resources"
+    }
+}
+
+public protocol SAMResource: Encodable {}
+public protocol SAMResourceProperties: Encodable {}
+
+public struct Resource: SAMResource, Equatable {
+    
+    let type: String
+    let properties: SAMResourceProperties
+    let name: String
+        
+    public static func none() -> [Resource] { return [] }
+    
+    public static func == (lhs: Resource, rhs: Resource) -> Bool {
+        lhs.type == rhs.type && lhs.name == rhs.name
+    }
+    
+    enum CodingKeys: String, CodingKey {
+        case type = "Type"
+        case properties = "Properties"
+    }
+    
+    // this is to make the compiler happy : Resource now confoms to Encodable
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try? container.encode(self.type, forKey: .type)
+        try? container.encode(self.properties, forKey: .properties)
+    }
+}
+
+//MARK: Lambda Function resource definition
+/*---------------------------------------------------------------------------------------
+ Lambda Function
+ 
+ https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-resource-function.html
+-----------------------------------------------------------------------------------------*/
+
+extension Resource {
+    public static func serverlessFunction(name: String,
+                                          codeUri: String,
+                                          eventSources: [EventSource] = [],
+                                          environment: EnvironmentVariable = EnvironmentVariable.none()) -> Resource {
+        
+        let properties = ServerlessFunctionProperties(codeUri: codeUri,
+                                                       eventSources: eventSources,
+                                                       environment: environment)
+        return Resource(type: "AWS::Serverless::Function",
+                        properties: properties,
+                        name: name)
+    }
+
+}
+
+public struct ServerlessFunctionProperties: SAMResourceProperties {
+    public enum Architectures: String, Encodable {
+        case x64 = "x86_64"
+        case arm64 = "arm64"
+    }
+    let architectures: [Architectures]
+    let handler: String
+    let runtime: String
+    let codeUri: String
+    let autoPublishAlias: String
+    var eventSources: [String: EventSource]
+    var environment: EnvironmentVariable
+    
+    public init(codeUri: String,
+                eventSources: [EventSource] = [],
+                environment: EnvironmentVariable = EnvironmentVariable.none()) {
+        
+        #if arch(arm64) //when we build on Arm, we deploy on Arm
```

Comment:
this is probably okay for now, but may not always be true

Reply:
I made the architecture a parameter in `Deploy.swift`, with default to the current architecture.
https://github.com/sebsto/swift-aws-lambda-runtime/commit/f3f4021dbebb086f4f33efac58b3b4e6a5cf4d1e